### ISSUE TITLE
ft_sourceanalysis always warns about sub-cfg specifications

### ISSUE THREAD

**Issue by britta-wstnr** [fieldtrip/fieldtrip#1453]
**Deprecation warning for both `cfg.reducerank` and `cfg.lcmv.reducerank`**
`ft_sourceanalysis` always throws a deprecation warning, no matter where the option `reducerank` is placed (`cfg` or `cfg.lcmv`):

```
Warning: The field cfg.lcmv.reducerank is deprecated, pleae use cfg.reducerank
```
If specified as `cfg.reducerank`, it _additionally_ throws:
```
Warning: The field cfg.reducerank is deprecated, pleae use cfg.lcmv.reducerank
```
(Typo in warning sic.)

**To Reproduce**
Run `ft_sourceanalysis` with `cfg.reducerank = 'no' ` or `cfg.lcmv.reducerank = 'no' `.

**Expected behavior**
Not sure which one should be deprecated.

**Reason**
Possibly an interaction between https://github.com/fieldtrip/fieldtrip/blob/master/ft_sourceanalysis.m#L215
where everything gets moved into the sub-cfg structure and 
https://github.com/fieldtrip/fieldtrip/blob/master/utilities/ft_checkconfig.m#L321
where the opposite is demanded as well as 
https://github.com/fieldtrip/fieldtrip/blob/master/ft_sourceanalysis.m#L217
where things get moved again.

This should also happen for other source analysis methods than LCMV as well as the other options specified in the list in `ft_checkconfig`

**Comment by schoffelen** [comment]
> 
> (Typo in warning sic.)
> 

Good stuff. The rest I don't understand either :)

**Comment by robertoostenveld** [comment]
This is due to a change that I recently made. The leadfield is not inverse-method specific, hence the leadfield options (like reducerank) should not be under cfg.method, but at the top-level.

The options that are inverse method specific should be in cfg.method, i.e. at the sub-level. 

There is code in ft_checkconfig that moves options from the top-level to the sub-level and vice versa. I recently made some changes here, but apparently they now get moved back-and-forth.

I will fix it.
